### PR TITLE
Add merchant to database after signing up

### LINKED ISSUE #72: Send Merchant Sign Up form data to Add Merchant API endpoint

### PR DESCRIPTION
**Pull request by karenfrilya97**
Closes #72

Merge after #79 and #81

### CODE REVIEW

File: server/.env [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# These are environment variables for development in local machine.
```
Comment (hzxcaryn) [inline server/.env:1]: If that's the case should it be `.env.development` or `.env.development.local` instead?

File: web/src/interfaces.ts [hunk 2]
```diff
@@ -64,3 +64,22 @@ export interface Customer {
...
+ * would be sent to the server to add the merchant to the database.
+ */
+export interface MerchantPayload {
+  name: string;
+  email: string;
+  vpa: string;
+}
+
+/**
+ * MerchantResponse Interface that contains the fields of the Response that
```
Comment (xlai20) [inline web/src/interfaces.ts:79]: Doc change: ...Interface that contains the fields of the response that client side would receive after merchant signup.

File: web/src/interfaces.ts [hunk 3]
```diff
@@ -64,3 +64,22 @@ export interface Customer {
...
+ * client side would receive.
+ */
+export interface MerchantResponse extends MerchantPayload {
```
Comment (xlai20) [inline web/src/interfaces.ts:82]: Probably the name "addMerchantResponse" is more appropriate to how you attempt to use it?

File: server/.env (hunk 1 above)
Comment (karenfrilya97) [reply server/.env:1]: Ohh cos dotenv cannot read `.env.development` :/

Comment (karenfrilya97) [reply server/.env:1]: Okay actually dotenv-flow can read `.env.development`.

File: web/src/api.ts [hunk 4]
```diff
@@ -33,3 +35,34 @@ export const getAllListings = async (): Promise<Listing[]> => {
...
+    `${process.env.REACT_APP_SERVER_URL}/merchants`,
+    requestOptions
+  );
+
+  if (res.status !== 201) {
+    throw new Error(Errors.SERVER_ERROR);
+  }
+
+  const location = res.headers.get('Location');
+  const merchantId = location?.substring(location?.lastIndexOf('/') + 1);
```
Comment (ifhuang) [inline web/src/api.ts:64]: should we add an extra field to res to return merchant id?
extracting from location is not very intuitive

Comment (karenfrilya97) [reply web/src/api.ts:64]: Yep! Actually that sounds good. I will make another PR for that?

File: web/src/interfaces.ts (hunk 2 above)
Comment (karenfrilya97) [reply web/src/interfaces.ts:79]: It's not the response that client side would receive after merchant signup for now. But I changed it accordingly to make it clearer.

File: web/src/interfaces.ts (hunk 3 above)
Comment (karenfrilya97) [reply web/src/interfaces.ts:82]: This response is supposed to be a representation for a Merchant object that is used in the HTTP response. So I think `MerchantResponse` would be more accurate.

**Pull request merged**